File: C++/Ohne Polymorphismus/OhnePolymorphismus.cpp
// Polymorphismus.cpp : Diese Datei enthält die Funktion "main". Hier beginnt und endet die Ausführung des Programms.
//

#include <iostream>
#include <list>
#include "Haustier.h"
#include "HaustierArt.h"



int main()
{
    Haustier* h1 = new Haustier(HaustierArt::HUND, "Bello");
    Haustier* h2 = new Haustier(HaustierArt::KATZE, "Mietze");
    Haustier* h3 = new Haustier(HaustierArt::WELLENSITTICH, "Hansi");
    Haustier* h4 = new Haustier(HaustierArt::HUND, "Lopi");

    list<Haustier*> haustiere;
    haustiere.push_back(h1);
    haustiere.push_back(h2);
    haustiere.push_back(h3);
    haustiere.push_back(h4);

    for (auto haustier : haustiere)
    {
        haustier->schreibeName();
        haustier->fressen();
        haustier->freuen();
        haustier->lautGeben();
        cout << endl;
    }
}

File: TypeScript/Ohne Polymorphismus/OhnePolymorphismus.ts
import {Haustier} from "./Haustier";
import {HaustierArt} from "./HaustierArt";


class Polymorphismus{
    static hauptprogramm(){
        let h1 : Haustier = new Haustier(HaustierArt.HUND, "Bello");
        let h2 : Haustier = new Haustier(HaustierArt.KATZE, "Mietze");
        let h3 : Haustier = new Haustier(HaustierArt.WELLENSITTICH, "Hansi");
        let h4 : Haustier = new Haustier(HaustierArt.HUND, "Lopi");

        let haustiere : Haustier[] = [h1, h2, h3, h4];

        for (let tier of haustiere){
            tier.schreibeName();
            tier.fressen();
            tier.freuen();
            tier.lautGeben();
            console.log("--------------");
        }
    }
}

Polymorphismus.hauptprogramm();